src/main/java/com/google/devtools/build/lib/bazel/coverage/CoverageReportActionBuilder.java

```diff
@@ -245,9 +247,13 @@ private CoverageReportAction generateCoverageReportAction(
     args = CoverageArgs.createCopyWithCoverageDirAndLcovOutput(args, coverageDir, lcovOutput);
     ImmutableList<String> actionArgs = argsFunc.apply(args);
 
-    ImmutableList<Artifact> inputs = ImmutableList.<Artifact>builder()
+    NestedSetBuilder<Artifact> inputsBuilder = NestedSetBuilder.stableOrder();
```

Comment:
What are the benefits of using a NestedSet instead of a list here? There are only a few artifacts, creating two NestedSets seems unnecessary.

Reply:
There ... aren't. Admittedly I'm not a Java developer, so I was just ripping off what I found in `TestActionBuilder`. I'll remove the `NestedSet` and change this to a seemingly simple one-line diff.

Reply:
(I think I assumed that `addTransitive` did something special w/r/t handling of the special "middleman" artifact type.)